keybase/platform_windows.go

```diff
@@ -58,9 +63,82 @@ func (c context) BeforeUpdatePrompt(update updater.Update, options updater.Updat
 	return nil
 }
 
+func (c config) promptProgram() (command.Program, error) {
+	destinationPath := c.destinationPath()
+	if destinationPath == "" {
+		return command.Program{}, fmt.Errorf("No destination path")
+	}
+
+	return command.Program{
+		Path: "mshta.exe",
+		Args: []string{filepath.Join(destinationPath, "prompter", "prompter.hta")},
+	}, nil
+}
+
+type updaterWinPromptInput struct {
+	Title       string `json:"title"`
+	Message     string `json:"message"`
+	Description string `json:"description"`
+	AutoUpdate  bool   `json:"autoUpdate"`
+	OutPathName string `json:"outPathName"`
+	TimeoutSecs int    `json:"timeoutSecs"`
+}
+
 func (c context) UpdatePrompt(update updater.Update, options updater.UpdateOptions, promptOptions updater.UpdatePromptOptions) (*updater.UpdatePromptResponse, error) {
-	// No update prompt for Windows, since the installer may handle it
-	return &updater.UpdatePromptResponse{Action: updater.UpdateActionContinue}, nil
+	promptProgram, err := c.config.promptProgram()
+	if err != nil {
+		return nil, err
+	}
+
+	promptJSONInput, err := c.promptInput(update, options, promptOptions)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error generating input: %s", err)
+	}
+
+	// Unmarshal so we can add a couple of fields
+	var promptArgs updaterPromptInput
+	err = json.Unmarshal([]byte(promptJSONInput), &promptArgs)
+	if err != nil {
+		return nil, fmt.Errorf("Error generating input while unmarshaling: %s", err)
+	}
+
+	tmpPathname := filepath.Join(os.TempDir(), fmt.Sprintf("updatePrompt%d.txt", rand.Intn(1000)))
+	defer os.Remove(tmpPathname)
+
+	promptJSONWinInput, err := json.Marshal(updaterWinPromptInput{
+		Title:       promptArgs.Title,
+		Message:     promptArgs.Message,
+		Description: promptArgs.Description,
+		AutoUpdate:  promptArgs.AutoUpdate,
+		OutPathName: tmpPathname,
+		TimeoutSecs: 3600, // to match time.Hour, below
+	})
+
+	_, err = command.Exec(promptProgram.Path, promptProgram.ArgsWith([]string{string(promptJSONWinInput)}), time.Hour, c.log)
+	if err != nil {
+		return nil, fmt.Errorf("Error running command: %s", err)
+	}
+
+	result, err := c.updaterPromptResultFromFile(tmpPathname)
+	if err != nil {
+		return nil, err
+	}
+	return c.responseForResult(*result)
+}
+
+// updaterPromptResultFromFile gets the result from the temp file and decodes it
+func (c context) updaterPromptResultFromFile(Name string) (*updaterPromptInputResult, error) {
+	resultRaw, err := ioutil.ReadFile(Name)
```

Comment:
use util.ReadFile(...)   (go-updater/util) ... util stuff is meant to be "safer" versions of these things